src/main/java/com/bobocode/hoverla/bring/context/BeanValidator.java

```diff
@@ -1,18 +1,125 @@
 
     /**
-     * @throws NoUniqueBeanException if there's bean duplicate found
+     * @param beanDefinitions all bean definitions of application context
+     * @throws IllegalArgumentException    if beanDefinitions is not initialized
+     * @throws NoUniqueBeanException       if there's bean duplicate found
+     * @throws IncorrectBeanNameException  if bean name is empty or blank
+     * @throws IncorrectBeanTypeException  if bean type is not suitable for the required type
+     * @throws CircularDependencyException if occurs circular dependency
+     * @throws NoSuchBeanException         if the application context doesn't contain such a bean
      */
     public void validate(List<BeanDefinition> beanDefinitions) {
+        log.debug("Bean validation started");
+        if (beanDefinitions == null) {
+            throw new IllegalArgumentException("Bean definition list is not initialized");
+        }
 
+        beanDefinitions.forEach(beanDefinition -> {
+            beanNameValidation(beanDefinition);
+            dependenciesValidation(beanDefinition, beanDefinitions);
+        });
+        var duplicateNames = beanDefinitions.stream()
+                .collect(Collectors.collectingAndThen(
+                        Collectors.groupingBy(BeanDefinition::name, Collectors.counting()),
+                        m -> m.entrySet().stream()
+                                .filter(v -> v.getValue() > 1)
+                                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue))
+                ));
+
+        if (!duplicateNames.isEmpty()) {
+            throw new NoUniqueBeanException("Context contains beans with duplicate names: " + duplicateNames.keySet());
+        }
+    }
+
+    private void beanNameValidation(BeanDefinition beanDefinition) {
+        log.trace("Checking name for bean definition: {} - {}", beanDefinition.name(), beanDefinition.type());
+        String name = beanDefinition.name();
+        if (beanDefinition.name() == null || name.isEmpty() || name.contains(" ")) {
+            throw new IncorrectBeanNameException(String.format("Bean name for %s class is null, empty or contains spaces",
+                    beanDefinition.type().getSimpleName()));
+        }
+    }
+
+    private void dependenciesValidation(BeanDefinition beanDefinition,
+                                        List<BeanDefinition> allBeanDefinitions) {
+        log.trace("Checking dependencies for bean definition: {} - {}", beanDefinition.name(),
+                beanDefinition.type());
+        if (beanDefinition.dependencies().isEmpty()) {
+            return;
+        }
+        var dependencies = beanDefinition.dependencies();
+
+        for (Map.Entry<String, Class<?>> dependency : dependencies.entrySet()) {
+            log.trace("***Checking dependency {} - {}", dependency.getKey(), dependency.getValue());
+            var allBeansByDependencyName = findAllBeanByName(dependency.getKey(), allBeanDefinitions);
```

Comment:
`var` is not aplicable here. It should be used when the return type is obvious (for example with `new` statement)

Reply:
Done